cartography/intel/gcp/compute.py

```diff
@@ -0,0 +1,118 @@
+# Google Compute Engine API-centric functions
+# https://cloud.google.com/compute/docs/concepts
+from googleapiclient.discovery import HttpError
+import json
+import logging
+
+from cartography.util import run_cleanup_job
+from cartography.intel.gcp.crm import get_gcp_projects
+
+logger = logging.getLogger(__name__)
+
+
+def get_zones_in_project(project_id, compute, max_results=None):
+    """
+    Return the zones where the Compute Engine API is enabled for the given project_id.
+    If the API is not enabled, return None.
+    :param project_id: The project id
+    :param compute: The compute resource object created by googleapiclient.discovery.build()
+    :param max_results: Optional cap on number of results returned by this function. Default = None, which means no cap.
+    :return: List of a project's zone objects if Compute API is turned on, else None.
+    """
+    try:
+        req = compute.zones().list(project=project_id, maxResults=max_results)
+        res = req.execute()
+        return res['items']
+    except HttpError as e:
+        reason = json.loads(e.content)['error']['errors'][0]['reason']
```

Comment:
are we sure the structure will always be present? We might spin up exception inside the handler

Reply:
+1 -- does the library not provide a better interface for getting this information?

Reply:
True.  I can change to `res.get('items', [])`

Reply:
From google SDK:

```python
    try:
      data = json.loads(self.content.decode('utf-8'))
      if isinstance(data, dict):
        reason = data['error']['reason']
      elif isinstance(data, list) and len(data) > 0:
        first_error = data[0]
        reason = first_error['error']['reason']
    except (ValueError, KeyError, TypeError):
        pass
```